ui/shared/components/panel/variants/VariantGene.jsx

```diff
@@ -34,12 +34,19 @@ const PADDED_INLINE_STYLE = {
   ...INLINE_STYLE,
 }
 
-const BaseGeneLabelContent = styled(({ color, customColor, label, maxWidth, containerStyle, dispatch, ...props }) => {
+const BaseGeneLabelContent = styled(({
+  color, customColor, hint, label, maxWidth, containerStyle, dispatch, ...props
+}) => {
   const labelProps = {
     ...props,
     size: 'mini',
     content: label,
   }
+
+  if (hint && customColor) {
```

Comment:
so I think you misunderstood what I meant by using the `detail` prop for the label. If you pass through a prop named `detail` directly to the label (pass it directly to `GeneDetailSection` and it will get passed through) and then it will render inside the label in its own div. If you use that prop for the initials, you can use this styled component to style that detail separately from the rest of the label content. You will not need to add any new components or conditional rendering. The easiest way I found to do this is to wrap the main label content in a span in the line above this (`content: label` -> `content: <span>{label}</span>`) and then change this styling below to the following:
```
margin: ${props => props.margin || '0px .5em .8em 0px'} !important;
white-space: nowrap;
   
span {
     display: inline-block;
     overflow: hidden;
     text-overflow: ellipsis;
     max-width: ${props => props.maxWidth || 'none'};
}

.detail {
     margin-left: 0.5em !important;
     
     &::before {
      content: "(";
     }
     &::after {
      content: ")";
     }
 }
```
This means the max width and overflow are applied to the label content and not the initials, which is I think better behavior (although it means we probably want a small value for max width here, maybe 9em)

Reply:
Oh! I didn't realise you could pass in selectors in StyledComponents. I couldn't figure out how to target the Label.Detail when I tried it before, but yes, this is much cleaner.

I also like using the before/after content to show the (), that works much better with the "hint".

I've cleaned the FlexLabel component to use `Label.Detail` and a normal `span`, and added a simple render test.

Reply:
So I think you misuderstood what I was saying. You should be adding no new styled components as part of this PR. The `FlexLabel` is unnecessary, the code I posted will work with no new components and was intended to be put here, in this already existing  `BaseGeneLabelContent` component. I did describe it in a few pieces which was maybe unclear, here is the full version of what the code should look like:

```
const BaseGeneLabelContent = styled(({ color, customColor, label, maxWidth, containerStyle, dispatch, hint, ...props })`
 const labelProps = {
    ...props,
    size: 'mini',
    content: <span>label<span>, // update here to wrap the content so it can be styled
    detail: hint, // add this to pass the correctly named prop to the label. I think it would be cleaner to just name the prop "detail' rather than rename it here, but this will work
 }
 return customColor ?
    <ColoredLabel {...labelProps} color={customColor} /> : <Label {...labelProps} color={color || 'grey'} />
 })`
  margin: ${props => props.margin || '0px .5em .8em 0px'} !important;
  white-space: nowrap;
   
  span {
     display: inline-block;
     overflow: hidden;
     text-overflow: ellipsis;
     max-width: ${props => props.maxWidth || 'none'};
  }

  .detail {
     margin-left: 0.5em !important;
     
     &::before {
      content: "(";
     }
     &::after {
      content: ")";
     }
 }
`
```

Reply:
Ah sorry, I didn't realise passing it in as a prop would work like that.

In that case, renaming hint to detail makes more sense, I was avoiding the word detail when I made the other components, that didn't use the native detail.

I've used your code, made a few fixes, and removed the extra component.